vetomint/src/lib.rs

```diff
@@ -120,18 +148,72 @@ pub struct HeightInfo {
     pub consensus_params: ConsensusParams,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+enum ConsensusStep {
+    Initial,
+    Propose,
+    Prevote,
+    Precommit,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+struct Votes {
+    prevotes_total: VotingPower,
+    prevotes_favor: BTreeMap<BlockIdentifier, VotingPower>,
+    /// If on-5f-prevotes has ever been triggered?
+    triggered_5f_prevote: bool,
```

Comment:
Why do we need this?
if 5f prevote is triggered, the consensus step is changed to Precommit, so it will not be retriggered.

Reply:
Then why do we have `for the first time` in the pseudo code? It is conditioned by `while step = prevote` anyway.